core/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java

```diff
@@ -515,16 +525,23 @@ public final ReducedQueryPhase reducedQueryPhase(List<? extends AtomicArray.Entr
             }
         }
         final Suggest suggest = groupedSuggestions.isEmpty() ? null : new Suggest(Suggest.reduce(groupedSuggestions));
+        ReduceContext reduceContext = new ReduceContext(bigArrays, scriptService, true);
         final InternalAggregations aggregations = aggregationsList.isEmpty() ? null : reduceAggs(aggregationsList,
-            firstResult.pipelineAggregators());
+            firstResult.pipelineAggregators(), reduceContext);
         final SearchProfileShardResults shardResults = profileResults.isEmpty() ? null : new SearchProfileShardResults(profileResults);
         return new ReducedQueryPhase(totalHits, fetchHits, maxScore, timedOut, terminatedEarly, firstResult, suggest, aggregations,
             shardResults);
     }
 
+
+    private InternalAggregations reduceAggsOnly(List<InternalAggregations> aggregationsList) {
```

Comment:
maybe update the name or add a comment to say that this method is about performing an intermediate reduce? (as opposed to final)